### ISSUE TITLE
Accept output of commans.do_task()

### ISSUE THREAD

**Issue by undg** [sourcegraph/sg.nvim#165]
I like to accept do_task() in same way as :CodyTask with <cr> or :CodyTaskAccept

Example:
```lua
local function generate_commit_message()
    local git_diff = vim.fn.system('git diff --cached')

    vim.api.nvim_command('G commit')

    commands.do_task(
        vim.api.nvim_get_current_buf(),
        0,
        1,
        'Suggest an informative commit message by summarizing code changes from the shared command output. The commit message should provide meaningful context for future readers.'
        .. git_diff
    )
end
```

It's do pick first line in commit buffer (I got there custom from git hook), however it doesn't want to replace it with generated text.

It there any way around it?

**Comment by undg** [comment]
After reading this part I wonder how I can get access to `M.tasks` and `M.active_task_index`


https://github.com/sourcegraph/sg.nvim/blob/6c592e9e78e68cd2bf4385da1b2a633219a22aab/after/plugin/cody.lua#L66-L71

Can I require anything from `/after/plugin/cody.lua` 🤔

**Comment by zippeurfou** [comment]
@undg have you solved your issue?
I am facing the same issue.

**Comment by undg** [comment]
Yeah! It's a bit hack'y and perhaps overkill, but I've followed similar approach that was done in tests.

Here you have few examples of how I've solved it:
https://github.com/undg/.dot/tree/master/vim/.config/nvim/lua/custom/sg/

And here I'm using those functions in keymaps and commands:
https://github.com/undg/.dot/blob/master/vim/.config/nvim/lua/plugins/sg.lua#L17-L36
